lib/src/main/kotlin/com/swmansion/starknet/provider/rpc/JsonRpcProvider.kt

```diff
@@ -153,17 +153,17 @@ class JsonRpcProvider(
         return callContract(call, BlockTag.LATEST)
     }
 
-    override fun deployAccount(payload: DeployAccountTransactionV1Payload): HttpRequest<DeployAccountResponse> {
-        val params = jsonWithDefaults.encodeToJsonElement(payload)
+    override fun deployAccount(payload: DeployAccountTransactionV1): HttpRequest<DeployAccountResponse> {
+        val params = jsonWithDefaults.encodeToJsonElement(TransactionSerializer, payload)
```

Comment:
Nit: 

Please check this, but afaik this shouldn't be necessary if you previously annotated `Transaction` class with `@Serializable(with = ...)`

Reply:
I didn't annotate transaction classes with `@Serializable(with = ...)` because it would create a recursive calls (`serialize` in `TransactionSerializer` would use serializer of specific transaction, which would then go back to `TransactionSerializer` and so on).

Also, transaction classes which inherit from `Transaction` aren't automatically serialized with `TransactionSerializer` even though there's such annotation (not sure why tbh):
```kotlin
@Serializable(with = TransactionSerializer::class)
sealed class Transaction : StarknetResponse {
```